### PR TITLE
APPSRE-4225: Improve Exception handling

### PR DESCRIPTION
**Pull request by lechuk47**
Improved two aspects: 
* Set `encoding` to `subprocess.run()` to avoid calling decode. 
* Add generic Exception handling. CalledProcessError is only raised if the process could run. If, for example, `jenkins-jobs` executable file does not exist, a `FileNotFoundException` is raised instead of CPE.

### CODE REVIEW

**Review by jfchevrette — APPROVED** [review]
LGTM!

**Comment by maorfr** [comment]
please try to use your fork :)

File: reconcile/utils/jjb_client.py [hunk 1]
```diff
@@ -238,13 +238,16 @@ def update(self):
                 result = subprocess.run(cmd,
                                         check=True,
                                         stdout=PIPE,
-                                        stderr=STDOUT)
-                out_str = result.stdout.decode("utf-8")
-                if re.search("updated: [1-9]", out_str):
-                    logging.info(out_str)
+                                        stderr=STDOUT,
+                                        encoding="utf-8")
+                if re.search("updated: [1-9]", result.stdout):
```
[the remark this answers is not in the document]
Comment (lechuk47) [reply reconcile/utils/jjb_client.py:243]: Added the type hint, I tried it and it works as expected. Thanks.

File: reconcile/utils/jjb_client.py [hunk 2]
```diff
@@ -238,13 +238,16 @@ def update(self):
...
+                    logging.info(result.stdout)
             except CalledProcessError as ex:
-                msg = ex.stdout.decode("utf-8")
-                logging.error(msg)
+                logging.exception(ex.stdout)
+                raise
```
[the remark this answers is not in the document]
Comment (lechuk47) [reply reconcile/utils/jjb_client.py]: Exactly. I changed to logging.error to just log the stdout attr and let run-integration to log the exception.

**Review by steveteahan — APPROVED** [review]

File: reconcile/utils/jjb_client.py [hunk 3]
```diff
@@ -238,13 +238,13 @@ def update(self):
...
-                if re.search("updated: [1-9]", out_str):
-                    logging.info(out_str)
+                                        stderr=STDOUT,
+                                        encoding="utf-8")
+                if re.search("updated: [1-9]", result.stdout):
+                    logging.info(result.stdout)
             except CalledProcessError as ex:
-                msg = ex.stdout.decode("utf-8")
-                logging.error(msg)
+                logging.error(ex.stdout)
```
Comment (Piojo) [inline reconcile/utils/jjb_client.py:246]: Why bother? Why not just remove the `try` block here and let somebody up the stack do the logging?

Comment (lechuk47) [reply reconcile/utils/jjb_client.py:246]: CalledProcessError stdout attribute contains the output of the failed command and this data is not printed with the base exception handling. I think it's worth to log the command's output and not just `command returned with non-zero error`. The exception is raised and managed up in the stack though.

Comment (Piojo) [reply reconcile/utils/jjb_client.py:246]: Fair enough. I'm convinced.

**Review by Piojo — APPROVED** [review]

**Pull request merged**